adapters/apigateway-domain-name.go

```diff
@@ -0,0 +1,233 @@
+		OwnershipVerificationCertificateArn: output.OwnershipVerificationCertificateArn,
+		RegionalCertificateName:             output.RegionalCertificateName,
+		RegionalCertificateArn:              output.RegionalCertificateArn,
+	}
+}
+
+func domainNameOutputMapper(_, scope string, awsItem *types.DomainName) (*sdp.Item, error) {
+	attributes, err := adapterhelpers.ToAttributesWithExclude(awsItem, "tags")
+	if err != nil {
+		return nil, err
+	}
+
+	item := sdp.Item{
+		Type:            "apigateway-domain-name",
+		UniqueAttribute: "DomainName",
+		Attributes:      attributes,
+		Scope:           scope,
+	}
+
+	// Health based on the DomainNameStatus
+	switch awsItem.DomainNameStatus {
+	case types.DomainNameStatusAvailable:
+		item.Health = sdp.Health_HEALTH_OK.Enum()
+	case types.DomainNameStatusUpdating,
+		types.DomainNameStatusPending,
+		types.DomainNameStatusPendingCertificateReimport,
+		types.DomainNameStatusPendingOwnershipVerification:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	default:
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_OTHER,
+			ErrorString: fmt.Sprintf("unknown Domain Name State: %s", awsItem.DomainNameStatus),
+		}
+	}
+
+	if awsItem.RegionalHostedZoneId != nil {
+		//+overmind:link route53-hosted-zone
+		item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+			Query: &sdp.Query{
+				Type:   "route53-hosted-zone",
+				Method: sdp.QueryMethod_GET,
+				Query:  *awsItem.RegionalHostedZoneId,
+				Scope:  scope,
+			},
+			BlastPropagation: &sdp.BlastPropagation{
+				// Changing the hosted zone can affect the domain name
+				In: true,
+				// The domain name won't affect the hosted zone
+				Out: false,
+			},
+		})
+	}
+
+	if awsItem.DistributionHostedZoneId != nil {
+		//+overmind:link route53-hosted-zone
+		item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+			Query: &sdp.Query{
+				Type:   "route53-hosted-zone",
+				Method: sdp.QueryMethod_GET,
+				Query:  *awsItem.DistributionHostedZoneId,
+				Scope:  scope,
+			},
+			BlastPropagation: &sdp.BlastPropagation{
+				// Changing the hosted zone can affect the domain name
+				In: true,
+				// The domain name won't affect the hosted zone
+				Out: false,
+			},
+		})
+	}
+
+	if awsItem.CertificateArn != nil {
+		//+overmind:link acm-certificate
+		item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+			Query: &sdp.Query{
+				Type:   "acm-certificate",
+				Method: sdp.QueryMethod_GET,
+				Query:  *awsItem.CertificateArn,
+				Scope:  scope,
+			},
+			BlastPropagation: &sdp.BlastPropagation{
+				// They are tightly linked
+				In:  true,
+				Out: true,
+			},
+		})
+	}
+
+	if awsItem.RegionalCertificateArn != nil {
+		//+overmind:link acm-certificate
+		item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+			Query: &sdp.Query{
+				Type:   "acm-certificate",
+				Method: sdp.QueryMethod_GET,
```

Comment:
This will be `SEARCH` since it's an ARN

```suggestion
				Method: sdp.QueryMethod_SEARCH,
```

Reply:
I thought about that, but this source only supports [getting](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/acm/get-certificate.html) or [describing](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/acm/describe-certificate.html) a certificate by its ARN.
How are we going to create the GET method for this source if we support search by its ARN?
Maybe we should support `disable_search`?

Reply:
Ah I see what you mean, it's always going to be the ARN as the query for GET too since that's the only input allowed. In that case let's leave it as it is, but add ARN parsing to it so that we can extract the correct scope. That way we'll support linking to certs that are in a different account or region, like this: https://github.com/overmindtech/aws-source/blob/44d13fa1e1e016e9613b4490a13d60c93f830b71/adapters/ecs-task.go#L94-L100

Reply:
https://github.com/overmindtech/aws-source/pull/654/commits/2a5bd15ba503474b207f5e1fee7446d57654dc63